File: Documentation/5.4/Samples/python/ClientApi/Session/ClusterTransaction/Overview.py
from ravendb import SessionOptions, TransactionMode
from examples_base import ExamplesBase


class HowToQuery(ExamplesBase):
    def setUp(self):
        super().setUp()

    def test_sync(self):
        with self.embedded_server.get_document_store("ClusterTransactionOverviewSync") as store:
            # region open_cluster_session_sync
            with store.open_session(
                session_options=SessionOptions(
                    transaction_mode=TransactionMode.CLUSTER_WIDE  # Set mode to be cluster-wide
                    # Session will be single-node when either:
                    #   * Mode is not specified
                    #   * Explicitly set TransactionMode.SINGLE_NODE
                )
            ) as session:
                # endregion
                ...

File: Documentation/5.4/Samples/php/ClientApi/session/ClusterTransaction/Overview.php
<?php

use RavenDB\Documents\Session\SessionOptions;
use RavenDB\Documents\Session\TransactionMode;
use RavenDB\Samples\Infrastructure\DocumentStoreHolder;

class Overview
{
    public function howToQuery(): void
    {
        $store = DocumentStoreHolder::getStore();
        try {

            # region open_cluster_session_sync
            $sessionOptions = new SessionOptions();

            // Set mode to be cluster-wide
            $sessionOptions->setTransactionMode(TransactionMode::clusterWide());

            // Session will be single-node when either:
            //   * Mode is not specified
            //   * Explicitly set TransactionMode.SingleNode

            $session = $store->openSession($sessionOptions);
            try {
                //
            } finally {
                $session->close();
            }
            # endregion
        } finally {
            $store->close();
        }
    }
}
